docs/source/basic_usage.rst

```diff
@@ -0,0 +1,106 @@
+Basic Usage
+===========
+
+``pygeoprocessing`` uses GDAL to read and write all GDAL-supported file
+formats. The utility functions
```

Comment:
I've found the GDAL [raster data model](https://gdal.org/user/raster_data_model.html) and [vector data model](https://gdal.org/user/vector_data_model.html) pages to be helpful in understanding the basic concepts. Maybe link to those somewhere in this section?